node/bundle/bundle.go

```diff
@@ -49,47 +51,58 @@ func (b *BundleFetcher) Fetch(version Version, release, netw string) (path strin
 			return bundleFilePath, nil
 		}
 
-		logb.Warnf("invalid bundle %s: %s; refetching", bundleName, err)
+		log.Warnf("invalid bundle %s: %s; refetching", bundleName, err)
 	}
 
-	logb.Infof("fetching bundle %s", bundleFile)
+	log.Infof("fetching bundle %s", bundleFile)
 	if err := b.fetch(release, bundleBasePath, bundleFile, bundleHash); err != nil {
-		logb.Errorf("error fetching bundle %s: %s", bundleName, err)
+		log.Errorf("error fetching bundle %s: %s", bundleName, err)
 		return "", xerrors.Errorf("error fetching bundle: %w", err)
 	}
 
 	if err := b.check(bundleBasePath, bundleFile, bundleHash); err != nil {
-		logb.Errorf("error checking bundle %s: %s", bundleName, err)
+		log.Errorf("error checking bundle %s: %s", bundleName, err)
 		return "", xerrors.Errorf("error checking bundle: %s", err)
 	}
 
 	return bundleFilePath, nil
 }
 
 func (b *BundleFetcher) fetchURL(url, path string) error {
-	logb.Infof("fetching URL: %s", url)
+	log.Infof("fetching URL: %s", url)
+
+	for i := 0; i < 3; i++ {
+		resp, err := http.Get(url) //nolint
+		if err != nil {
+			if isTemporary(err) {
+				log.Warnf("temporary error fetching %s: %s; retrying in 1s", url, err)
+				time.Sleep(time.Second)
+				continue
+			}
+			return xerrors.Errorf("error fetching %s: %w", url, err)
+		}
+		defer resp.Body.Close() //nolint
 
-	resp, err := http.Get(url) //nolint
-	if err != nil {
-		return xerrors.Errorf("error fetching %s: %w", url, err)
-	}
-	defer resp.Body.Close() //nolint
+		if resp.StatusCode != http.StatusOK {
+			log.Warnf("unexpected response fetching %s: %s (%d); retrying in 1s", url, resp.Status, resp.StatusCode)
+			time.Sleep(time.Second)
+			continue
+		}
 
-	if resp.StatusCode != http.StatusOK {
-		return xerrors.Errorf("error fetching %s: http response status is %d", url, resp.StatusCode)
-	}
+		f, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+		if err != nil {
+			return xerrors.Errorf("error opening %s for writing: %w", path, err)
+		}
+		defer f.Close() //nolint
 
-	f, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
-	if err != nil {
-		return xerrors.Errorf("error opening %s for writing: %w", path, err)
-	}
-	defer f.Close() //nolint
+		if _, err := io.Copy(f, resp.Body); err != nil {
+			return xerrors.Errorf("error writing %s: %w", path, err)
+		}
```

Comment:
I wonder if we should add HTTP resumption support. More often than not what will happen is that the download gets interrupted due to a flaky connection. In that case, we don't want to truncate and restart the download from scratch, but rather resume it?

Reply:
How does go-paramfetcher do it?

Reply:
matbe, but it shouldnt be much of a proble. It is a single 30M download.

We can consider in a follow up if it turns out to be a problem.